pkg/services/ngalert/migration/store/database.go

```diff
@@ -226,62 +224,59 @@ var revertPermissions = []accesscontrol.Permission{
 // RevertAllOrgs reverts the migration, deleting all unified alerting resources such as alert rules, alertmanager configurations, and silence files.
 // In addition, it will delete all folders and permissions originally created by this migration, these are stored in the kvstore.
 func (ms *migrationStore) RevertAllOrgs(ctx context.Context) error {
-	return ms.store.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
-		if _, err := sess.Exec("DELETE FROM alert_rule"); err != nil {
-			return err
-		}
-
-		if _, err := sess.Exec("DELETE FROM alert_rule_version"); err != nil {
-			return err
-		}
+	return ms.store.InTransaction(ctx, func(ctx context.Context) error {
+		return ms.store.WithDbSession(ctx, func(sess *db.Session) error {
+			if _, err := sess.Exec("DELETE FROM alert_rule"); err != nil {
+				return err
+			}
 
-		orgs, err := ms.GetAllOrgs(ctx)
-		if err != nil {
-			return fmt.Errorf("get orgs: %w", err)
-		}
-		for _, o := range orgs {
-			if err := ms.DeleteMigratedFolders(ctx, o.ID); err != nil {
-				ms.log.Warn("Failed to delete migrated folders", "orgID", o.ID, "err", err)
-				continue
+			if _, err := sess.Exec("DELETE FROM alert_rule_version"); err != nil {
+				return err
 			}
-		}
 
-		if _, err := sess.Exec("DELETE FROM alert_configuration"); err != nil {
-			return err
-		}
+			orgs, err := ms.GetAllOrgs(ctx)
+			if err != nil {
+				return fmt.Errorf("get orgs: %w", err)
+			}
+			for _, o := range orgs {
+				if err := ms.DeleteMigratedFolders(ctx, o.ID); err != nil {
+					ms.log.Warn("Failed to delete migrated folders", "orgID", o.ID, "err", err)
+					continue
+				}
+			}
 
-		if _, err := sess.Exec("DELETE FROM ngalert_configuration"); err != nil {
-			return err
-		}
+			if _, err := sess.Exec("DELETE FROM alert_configuration"); err != nil {
+				return err
+			}
 
-		if _, err := sess.Exec("DELETE FROM alert_instance"); err != nil {
-			return err
-		}
+			if _, err := sess.Exec("DELETE FROM ngalert_configuration"); err != nil {
+				return err
+			}
 
-		if _, err := sess.Exec("DELETE FROM kv_store WHERE namespace = ?", notifier.KVNamespace); err != nil {
-			return err
-		}
+			if _, err := sess.Exec("DELETE FROM alert_instance"); err != nil {
+				return err
+			}
 
-		if _, err := sess.Exec("DELETE FROM kv_store WHERE namespace = ?", KVNamespace); err != nil {
-			return err
-		}
+			if _, err := sess.Exec("DELETE FROM kv_store WHERE namespace = ?", notifier.KVNamespace); err != nil {
+				return err
+			}
 
-		files, err := filepath.Glob(filepath.Join(ms.cfg.DataPath, "alerting", "*", "silences"))
-		if err != nil {
-			return err
-		}
-		for _, f := range files {
-			if err := os.Remove(f); err != nil {
-				ms.log.Error("Failed to remove silence file", "file", f, "err", err)
+			if _, err := sess.Exec("DELETE FROM kv_store WHERE namespace = ?", KVNamespace); err != nil {
+				return err
```

Comment:
On the quick glance this is not obvious that it basically does `SetMigrated(anyorg,false)` as well as per-org. It would be great if we mention it in the comment here or where it is called